lib/std/crypto/aes/aesni.zig

```diff
@@ -0,0 +1,420 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+// Based on Go stdlib implementation
+
+const std = @import("../../std.zig");
+const mem = std.mem;
+const debug = std.debug;
+const Vector = std.meta.Vector;
+
+const BlockVec = Vector(2, u64);
+
+/// A single AES block.
+pub const Block = struct {
+    pub const block_size: usize = 16;
+
+    /// Internal representation of a block.
+    repr: BlockVec,
+
+    /// Convert a byte sequence into an internal representation.
+    pub inline fn fromBytes(bytes: *const [16]u8) Block {
+        const repr = @ptrCast(*align(1) const BlockVec, bytes).*;
```

Comment:
Why not pass `bytes` by value and use `@bitCast`?

Reply:
This is `mem.bytesToValue`

Reply:
If my memory serves me correctly, using `@bitCast` generated `movdqa` instructions instead of `movdqu`. The input bytes may not be aligned.

Reply:
When trying to vectorise aes implementation i've seen a lot of strange instruction choices. Maybe LLVM knows better.

Reply:
Nothing strange here. A 128 bit vector is usually aligned, just like other types, so emitting `movqda` makes sense. But here, we're reading from a user-supplied pointer, that is not meant to be 16-byte aligned, so we explicitly need the unaligned version. On current CPUs, the difference between both is negligible anyway.

Reply:
Changed to `mem.bytesToValue()`, thanks @LemonBoy!